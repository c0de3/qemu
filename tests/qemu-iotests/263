#!/usr/bin/env python
#
# Test mirroring qcow2 under raw-format
#
# Copyright (c) 2019 Virtuozzo International GmbH.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#

import iotests
from iotests import qemu_img_create, qemu_io, filter_qemu_io, file_path, log

iotests.verify_image_format(supported_fmts=['qcow2'])

base, top, target = file_path('base', 'top', 'target')
size = 1024 * 1024

qemu_img_create('-f', iotests.imgfmt, base, str(size))
log(filter_qemu_io(qemu_io('-f', iotests.imgfmt,
                           '-c', 'write -P 1 0 1M', base)))
qemu_img_create('-f', iotests.imgfmt, '-b', base, top, str(size))

vm = iotests.VM().add_drive(None, opts='driver=raw,size=' + str(size) +
                            ',file.driver=qcow2,file.file.filename=' + top)
vm.launch()

vm.qmp_log('drive-mirror', device='drive0', target=target, sync='full',
           filters=[iotests.filter_qmp_testfiles])
log(vm.event_wait('BLOCK_JOB_READY'), filters=[iotests.filter_qmp_event])
vm.qmp_log('block-job-complete', device='drive0')
vm.shutdown()

log(filter_qemu_io(qemu_io('-f', 'raw', '-c', 'read -P 1 0 1M', target)))

log('-- finish --') # avoid extra new-line at the end of .out file
