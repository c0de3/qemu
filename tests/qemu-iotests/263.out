wrote 1048576/1048576 bytes at offset 0
1 MiB, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)

{"execute": "drive-mirror", "arguments": {"device": "drive0", "sync": "full", "target": "TEST_DIR/PID-target"}}
{"return": {}}
{"data": {"device": "drive0", "len": 1048576, "offset": 1048576, "speed": 0, "type": "mirror"}, "event": "BLOCK_JOB_READY", "timestamp": {"microseconds": "USECS", "seconds": "SECS"}}
{"execute": "block-job-complete", "arguments": {"device": "drive0"}}
{"return": {}}
read 1048576/1048576 bytes at offset 0
1 MiB, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)

-- finish --
